packages/vuetify/src/components/VDataTable/VDataTable.ts

```diff
@@ -458,7 +455,7 @@ export default VDataIterator.extend({
         key: getObjectValueByPath(item, this.itemKey),
         class: mergeClasses(
           { ...classes, 'v-data-table__selected': data.isSelected },
-          this.itemClass(item)
+          typeof this.itemClass === 'function' ? this.itemClass(item) : getObjectValueByPath(item, this.itemClass)
```

Comment:
use `getPropertyFromItem`, it accepts function as well so no need to checking if itemClass is function or string

Reply:
Oh, perfect, I change that! :)

Reply:
So, I restore the default the default value, as null value cause unexpected behavior now. But it works :)